clash_lib/src/proxy/utils/test_utils/mod.rs

```diff
@@ -42,26 +47,30 @@ pub async fn ping_pong_test(handler: Arc<dyn OutboundHandler>, port: u16) -> any
         let chunk = "world";
         let mut buf = vec![0; 5];
 
+        tracing::info!("destination_fn start read");
+
         for _ in 0..100 {
             read_half.read_exact(&mut buf).await?;
             assert_eq!(&buf, b"hello");
         }
 
+        tracing::info!("destination_fn start write");
+
         for _ in 0..100 {
             write_half.write_all(chunk.as_bytes()).await?;
             write_half.flush().await?;
         }
+
+        tracing::info!("destination_fn end");
         Ok(())
     }
 
     let target_local_server_handler = tokio::spawn(async move {
-        match listener.accept().await {
-            Ok((stream, _)) => destination_fn(stream).await,
-            Err(e) => {
-                // Handle error e, log it, or ignore it
-                error!("Failed to accept connection: {}", e);
-                Err(anyhow!("Failed to accept connection: {}", e))
-            }
+        loop {
```

Comment:
@VendettaReborn this should be a loop

Reply:
for our case it might be fine as we only have 1 conn.

Reply:
it was actually failing for me on macos as the proxy server inside container can't connect to the listener outside the vm on the host mac machine